sklearn/metrics/tests/test_score_objects.py

```diff
@@ -39,6 +40,10 @@
 from sklearn.model_selection import GridSearchCV
 from sklearn.multiclass import OneVsRestClassifier
 from sklearn.externals import joblib
+from functools import partial
+
+jaccard_similarity_score = partial(jaccard_similarity_score, average='binary')
```

Comment:
because of this redefinition, currently an error occurs:
```python
================================================================== FAILURES ===================================================================
____________________________________________________________ test_all_scorers_repr ____________________________________________________________

    def test_all_scorers_repr():
        # Test that all scorers have a working repr
        for name, scorer in SCORERS.items():
>           repr(scorer)

sklearn/metrics/tests/test_score_objects.py:158: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <[AttributeError("'functools.partial' object has no attribute '__name__'") raised in repr()] SafeRepr object at 0x7f2b38adc908>

    def __repr__(self):
        kwargs_string = "".join([", %s=%s" % (str(k), str(v))
                                 for k, v in self._kwargs.items()])
        return ("make_scorer(%s%s%s%s)"
>               % (self._score_func.__name__,
                   "" if self._sign > 0 else ", greater_is_better=False",
                   self._factory_args(), kwargs_string))
E       AttributeError: 'functools.partial' object has no attribute '__name__'

sklearn/metrics/scorer.py:68: AttributeError
============================================= 1 failed, 495 passed, 11 warnings in 25.33 seconds ==============================================
```

I can understand why it occurs, but I can't think of a good solution.

Reply:
Firstly, it might not be a good idea to overwrite the variable name. Secondly, `make_scorer` allows you to pass kwargs to it without partial.